packages/react-components/src/components/MessageThread.tsx

```diff
@@ -909,21 +909,22 @@ export const MessageThreadWrapper = (props: MessageThreadProps): JSX.Element =>
   const [readCountForHoveredIndicator, setReadCountForHoveredIndicator] = useState<number | undefined>(undefined);
 
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
-  const [inlineAttachments, setInlineAttachments] = useState<Record<string, string>>({});
+  const [inlineAttachments, setInlineAttachments] = useState<Record<string, Record<string, string>>>({});
   /* @conditional-compile-remove(teams-inline-images-and-file-sharing) */
   const onFetchInlineAttachment = useCallback(
-    async (attachment: FileMetadata): Promise<void> => {
-      if (!onFetchAttachments || attachment.attachmentType !== 'inlineImage' || attachment.id in inlineAttachments) {
+    async (attachments: FileMetadata[], messageId: string): Promise<void> => {
+      if (!onFetchAttachments) {
         return;
       }
+      const attachmentDownloadResult = await onFetchAttachments(attachments);
+      const listOfAttachments = inlineAttachments[messageId];
```

Comment:
Maybe do this instead if message id is not yet in the map
```
const listOfAttachments = inlineAttachments[messageId] ?? {}
```
So the `if` check in line 1183 can be removed